flask-app/app.py

```diff
@@ -0,0 +1,186 @@
+from flask import Flask, render_template, json, request, redirect, url_for
+from flaskext.mysql import MySQL
+import json
+import secrets
+
+mysql = MySQL()
+app = Flask(__name__)
+
+# MySQL configurations
+app.config['MYSQL_DATABASE_USER'] = secrets.user
+app.config['MYSQL_DATABASE_PASSWORD'] = secrets.password
+app.config['MYSQL_DATABASE_DB'] = secrets.db
+app.config['MYSQL_DATABASE_HOST'] = secrets.host
+mysql.init_app(app)
+
+
+def retrieveUnlabelledClips(count):
+  query = """ SELECT url, start, end FROM labelled
+              WHERE labelled = 0 ORDER BY RAND()
+              LIMIT {:} """.format(count)
+  try:
+    # connect to db 
+    conn = mysql.connect()
+    cur = conn.cursor()
+    # call select query
+    cur.execute(query)
+    data = cur.fetchall()
+    # close connection
+    cur.close()
+    conn.close()
+    return data
+  except Exception as e:
+    return { error_msg: 'Error occured retrieving clips', error: e }
+
+
+def formatYoutubeEmbedUrl(url, start, end):
+  embed_url = url.replace("watch?v=", "embed/")
+  params = "?start={:}&end={:}&autoplay=1".format(start, end)
+  return embed_url + params
+
+
+@app.route("/", methods=['GET'])
+def index():
+  if request.method == "GET":
+    clips = retrieveUnlabelledClips(1)
+    embed_url = ''
+    if len(clips) == 0:
+      return 'No more clips to label'
+    for clip in clips:
+      url, start, end = clip
+      embed_url = formatYoutubeEmbedUrl(url, start, end)
+    return render_template('index.html', embedUrl= embed_url, youtubeUrl= url, start= start, end= end)
+
+
+def formatAddLabelQuery(url, start, end, data):
+  commentator = data['commentator']
+  crowd = data['crowd']
+  gameplay = data['gameplay']
+
+  query = """ UPDATE labelled
+              SET
+                commentator = {:},
+                crowd = {:},
+                gameplay = {:},
+                labelled = 1,
```

Comment:
Now - I am not sure how we started using ints as booleans - but this is not best practice. I think I also neglected such a mistake in the other table. 

```
    gameplay    int                                null,
    labelled    int      default 0                 not null,
    updated_at  datetime default CURRENT_TIMESTAMP null,
    primary key (url, start, end)
);
```
It might be better for us to do ``` completed BOOLEAN ``` in the DDL - anyways I am just thinking out loud, but this will be a clean up boi ... one day.

Reply:
fixed 👍